timelock-corruption-detection/src/main/java/com/palantir/history/PaxosLogHistoryProvider.java

```diff
@@ -61,16 +80,131 @@ private Long getOrInsertVerificationState(NamespaceAndUseCase namespaceAndUseCas
                 namespaceAndUseCase.namespace(), namespaceAndUseCase.useCase());
     }
 
-    public List<CompletePaxosHistoryForNamespaceAndUseCase> getHistory() {
+    /**
+     * Gets history from all the nodes for all unique namespace, useCase tuples since last verified sequence numbers
+     *
+     * @throws  RuntimeException
+     *          if fails to fetch history from all remote servers
+     */
+    public List<CompletePaxosHistoryForNamespaceAndUseCase> getHistory() throws RuntimeException {
```

Comment:
nit: we don't generally declare unchecked exceptions
```suggestion
    public List<CompletePaxosHistoryForNamespaceAndUseCase> getHistory() {
```